packages/web-lib/src/persistence/web-store-manager.ts

```diff
@@ -0,0 +1,30 @@
+import { SessionStorageStrategy } from './strategies';
+import { IStorageStrategy, IWebStore, OperationResult } from './web-store.types';
+
+export class StorageContext implements IWebStore {
+  private strategy: IStorageStrategy;
+
+  constructor(strategy?: IStorageStrategy) {
+    this.strategy = strategy ?? new SessionStorageStrategy();
+  }
+
+  getStrategy(): IStorageStrategy {
+    return this.strategy;
+  }
+  setStrategy(strategy: IStorageStrategy): void {
```

Comment:
This is good! Additional improvement. We can pass strategy name instead of IStorageStrategy. 
Reasoning: we will expose only WebStoreManager aka StoreContext to users (only export from this module), and then users will be able to set strategy in run time.

const store = new WebStoreManager().setStrategy('indexdb')